simulation/traffic_simulator/include/traffic_simulator/data_type/entity_status.hpp

```diff
@@ -17,60 +17,69 @@
+using EntitySubtype = traffic_simulator_msgs::msg::EntitySubtype;
 
 inline namespace entity_status
 {
 class CanonicalizedEntityStatus
 {
 public:
-  explicit CanonicalizedEntityStatus(
-    const EntityStatus & may_non_canonicalized_entity_status,
-    const std::shared_ptr<hdmap_utils::HdMapUtils> & hdmap_utils);
   explicit CanonicalizedEntityStatus(
     const EntityStatus & may_non_canonicalized_entity_status,
     const std::optional<CanonicalizedLaneletPose> & canonicalized_lanelet_pose);
-  explicit CanonicalizedEntityStatus(
-    const EntityStatus & may_non_canonicalized_entity_status,
-    const std::shared_ptr<hdmap_utils::HdMapUtils> & hdmap_utils,
-    const lanelet::Ids & route_lanelets);
   explicit CanonicalizedEntityStatus(const CanonicalizedEntityStatus & obj);
   explicit operator EntityStatus() const noexcept { return entity_status_; }
-  auto operator=(const CanonicalizedEntityStatus & obj) -> CanonicalizedEntityStatus &;
+
+  auto set(
+    const EntityStatus & status, const double matching_distance,
+    const std::shared_ptr<hdmap_utils::HdMapUtils> & hdmap_utils_ptr) -> void;
+  auto set(const CanonicalizedEntityStatus & status) -> void;
   auto setAction(const std::string & action) -> void;
-  auto getName() const noexcept -> const std::string & { return entity_status_.name; };
-  auto getBoundingBox() const noexcept -> traffic_simulator_msgs::msg::BoundingBox;
-  auto laneMatchingSucceed() const noexcept -> bool;
+
+  auto getTime() const noexcept -> double;
+  auto setTime(double) -> void;
+
   auto getMapPose() const noexcept -> geometry_msgs::msg::Pose;
-  auto getLaneletPose() const -> LaneletPose;
-  auto getLaneletId() const -> lanelet::Id;
-  auto getLaneletIds() const -> lanelet::Ids;
-  auto getCanonicalizedLaneletPose() const -> std::optional<CanonicalizedLaneletPose>;
-  auto setTwist(const geometry_msgs::msg::Twist & twist) -> void;
+  auto setMapPose(const geometry_msgs::msg::Pose & pose) -> void;
+
   auto getTwist() const noexcept -> geometry_msgs::msg::Twist;
+  auto setTwist(const geometry_msgs::msg::Twist & twist) -> void;
   auto setLinearVelocity(double linear_velocity) -> void;
+
+  auto getAccel() const noexcept -> geometry_msgs::msg::Accel;
   auto setAccel(const geometry_msgs::msg::Accel & accel) -> void;
   auto setLinearAcceleration(double linear_acceleration) -> void;
-  auto getAccel() const noexcept -> geometry_msgs::msg::Accel;
-  auto setLinearJerk(double) -> void;
+
   auto getLinearJerk() const noexcept -> double;
-  auto setTime(double) -> void;
-  auto getTime() const noexcept -> double;
+  auto setLinearJerk(double) -> void;
+
+  auto laneMatchingSucceed() const noexcept -> bool;
+  auto getLaneletId() const noexcept -> lanelet::Id;
+  auto getLaneletIds() const noexcept -> lanelet::Ids;
+  auto getLaneletPose() const noexcept -> LaneletPose;
+  auto getCanonicalizedLaneletPose() const noexcept -> std::optional<CanonicalizedLaneletPose>;
```

Comment:
```suggestion
  auto getCanonicalizedLaneletPose() const noexcept -> const std::optional<CanonicalizedLaneletPose> &;
```

Reply:
@yamacir-kit 
Thank you so much, my overlook, I've [provided const reference where it was missing](https://github.com/tier4/scenario_simulator_v2/pull/1320/commits/850f40441c8727d8ca55705c9ff3ea4354edf680#diff-541c9fdcaf8e81f7ab6d1fb2a82521c2de7fb693b30cb8c466cadeedde341757) in `CanonicalizedEntityStatus`. :bow: